File: 김순석/프로그래머스/pro_예상대진표.py
def solution(n, a, b):
    answer = 0
    a, b = min(a, b), max(a, b)
    exp = 1
    tmp = b

    while tmp > 2:
        tmp //= 2
        exp += 1
    s = 1
    e = 2 ** exp
    print(e)
    while True:
        mid = (s + e) / 2
        if (mid - a) * (mid - b) < 0:
            answer = exp
            return answer
        else:
            exp -= 1
            if mid > b:
                e = int(mid)
            else:
                s = int(mid) + 1

    return answer

n, a, b = 16, 2, 8
print(solution(n, a, b))

File: AHYEON/13.programmers_lv2/pro_예상대진표.swift
func solution(_ n:Int, _ a:Int, _ b:Int) -> Int
{
    var A = a
    var B = b
    var round = 0

    while A != B {

        if A%2 == 1 {
            A = (A + 1)/2
        }else{
            A = A/2
        }

        if B%2 == 1 {
            B = (B + 1)/2
        }else{
           B = B/2
        }


        round += 1
    }
    return round
}
